code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -118,6 +118,136 @@ public static void updatePropositions(
+                            + " empty.");
+            AEPOptimizeError aepOptimizeError =
+                    new AEPOptimizeError(null, null, null, null, AdobeError.UNEXPECTED_ERROR);
+            failWithOptimizeError(callback, aepOptimizeError);
+            return;
+        }
+
+        final List<DecisionScope> validScopes = new ArrayList<>();
+        for (final DecisionScope scope : decisionScopes) {
+            if (!scope.isValid()) {
+                continue;
+            }
+            validScopes.add(scope);
+        }
+
+        if (validScopes.size() == 0) {
+            Log.warning(
+                    OptimizeConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Cannot update propositions, provided list of decision scopes has no valid"
+                            + " scope.");
+            return;
+        }
+
+        final List<Map<String, Object>> flattenedDecisionScopes = new ArrayList<>();
+        for (final DecisionScope scope : validScopes) {
+            flattenedDecisionScopes.add(scope.toEventData());
+        }
+
+        final Map<String, Object> eventData = new HashMap<>();
+        eventData.put(
+                OptimizeConstants.EventDataKeys.REQUEST_TYPE,
+                OptimizeConstants.EventDataValues.REQUEST_TYPE_UPDATE);
+        eventData.put(OptimizeConstants.EventDataKeys.DECISION_SCOPES, flattenedDecisionScopes);
+
+        if (!OptimizeUtils.isNullOrEmpty(xdm)) {
+            eventData.put(OptimizeConstants.EventDataKeys.XDM, xdm);
+        }
+
+        if (!OptimizeUtils.isNullOrEmpty(data)) {
+            eventData.put(OptimizeConstants.EventDataKeys.DATA, data);
+        }
+
+        final Event event =
+                new Event.Builder(
+                                OptimizeConstants.EventNames.UPDATE_PROPOSITIONS_REQUEST,
+                                OptimizeConstants.EventType.OPTIMIZE,
+                                OptimizeConstants.EventSource.REQUEST_CONTENT)
+                        .setEventData(eventData)
+                        .build();
+
+        MobileCore.dispatchEventWithResponseCallback(
+                event,
+                OptimizeConstants.GET_RESPONSE_CALLBACK_TIMEOUT,
+                new AdobeCallbackWithError<Event>() {
+                    @Override
+                    public void fail(final AdobeError adobeError) {
+                        AEPOptimizeError aepOptimizeError =
+                                new AEPOptimizeError(null, null, null, null, adobeError);
+                        failWithOptimizeError(callback, aepOptimizeError);
+                    }
+
+                    @Override
+                    public void call(final Event event) {
+                        try {
+                            final Map<String, Object> eventData = event.getEventData();
+                            if (OptimizeUtils.isNullOrEmpty(eventData)) {
+                                AEPOptimizeError aepOptimizeError =
```

Comment:
Since there are multiple scenarios where you create aepOptimizeError for adobeError like "Unexpected Error", "Invalid Request" etc, please create static function in AEPOptimizeError class like createAEPOptimizeUnexpectedError, createAEPOptimizeInvalidRequestError and define status, title and detail for these cases in OptimizeConstants and use them.

Reply:
Added these